### PR TITLE
feat: design types for the project charts

### PR DESCRIPTION
**Pull request by benfdking**

### CODE REVIEW

File: proto/quary/service/v1/chart.proto [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+option java_multiple_files = true;
+
+message Chart {
+  string name = 1;
+  optional string description = 2;
+  repeated string tags = 3;
+
+  oneof dependency {
+    ChartDependency path = 5;
```
Comment (louisjoecodes) [inline proto/quary/service/v1/chart.proto:13]: nit: we use 'path' for fully qualified path in other places might be confusing convention

File: proto/quary/service/v1/project_file.proto [hunk 2]
```diff
@@ -78,3 +81,23 @@ message ModelTest {
...
+  optional string description = 2;
+  repeated string tags = 3;
+
+  message AssetDependency {
+    string name = 1;
+    optional int64 limit = 2;
+  }
+
+  oneof source {
+    string sql = 4;
```
Comment (louisjoecodes) [inline proto/quary/service/v1/project_file.proto:98]: Why do we store the sql here?

**Comment by benfdking** [comment]
@louisjoecodes just rethinking this slightly. I think we should just have a chart file which is a 

'Name.chart.yaml' file that is defined in that one way. 

And that is a file that defines it.

File: proto/quary/service/v1/chart.proto [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import "google/protobuf/struct.proto";
+
+option java_multiple_files = true;
+
+// Parsed chart that is in a project
+message Chart {
+  // Name of the chart
+  string name = 1;
+  // Description of the cart
```
Comment (louisjoecodes) [inline proto/quary/service/v1/chart.proto:13]:
```suggestion
  // Description of the chart
```

File: proto/quary/service/v1/chart.proto [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    string pre_templated_sql = 5;
+    // reference to a pre-existing asset, model, source, seed, snapshot
+    AssetReference reference = 6;
+  }
+
+  // Configuration for the chart that is passed to perspective
+  google.protobuf.Struct config = 7;
+
+  // Dependencies that this chart has
+  repeated string dependencies = 8;
```
Comment (louisjoecodes) [inline proto/quary/service/v1/chart.proto:37]:
```suggestion
  repeated string references = 8;
```

**Review by louisjoecodes — APPROVED** [review]

**Pull request merged**